### PR TITLE
Add WebSocket Compression

### PR DESCRIPTION
**Pull request by BrennanConroy**
API docs don't have the APIs yet, so can't link to them

### CODE REVIEW

File: aspnetcore/fundamentals/websockets.md
Comment (guardrex) [inline aspnetcore/fundamentals/websockets.md:97]: We avoid gerunds in headings because they don't translate well. Can this be? ...

```suggestion
### Compression
```

File: aspnetcore/fundamentals/websockets.md
Comment (guardrex) [inline aspnetcore/fundamentals/websockets.md:100]: If it must be "CRIME/BREACH type attacks," then hyphenate it ("CRIME/BREACH-type"); otherwise ...

```suggestion
> Enabling compression over encrypted connections makes the app subject to CRIME/BREACH attacks.
```

File: aspnetcore/fundamentals/websockets.md
Comment (guardrex) [inline aspnetcore/fundamentals/websockets.md:101]: Code-fence API that we'll link later.

```suggestion
> If sending sensitive information, avoid enabling compression or use `WebSocketMessageFlags.DisableCompression` when calling `WebSocket.SendAsync`.
```

File: aspnetcore/fundamentals/websockets.md
Comment (guardrex) [inline aspnetcore/fundamentals/websockets.md:102]:
```suggestion
> This applies to both sides of the WebSocket. Note that the WebSockets API in the browser doesn't have configuration for disabling compression per send. 
```

File: aspnetcore/fundamentals/websockets.md
Comment (guardrex) [inline aspnetcore/fundamentals/websockets.md:104]: The suggestion will add a blank line between the sentence and the code block that follows:

```suggestion
If compression of messages over WebSockets is desired, then the accept code must specify that it allows compression as follows:

```

File: aspnetcore/fundamentals/websockets.md
Comment (guardrex) [inline aspnetcore/fundamentals/websockets.md:112]:
```suggestion
Additionally, `ServerMaxWindowBits` and `DisableServerContextTakeover` advanced options of <xref:Microsoft.AspNetCore.Http.WebSocketAcceptContext> control how the compression works.
```

File: aspnetcore/fundamentals/websockets.md
Comment (guardrex) [inline aspnetcore/fundamentals/websockets.md:114]:
```suggestion
Compression is negotiated between the client and server when first establishing a connection. You can read more about the negotiation in the [Compression Extensions for WebSocket RFC](https://datatracker.ietf.org/doc/html/rfc7692#section-7).
```

File: aspnetcore/fundamentals/websockets.md
Comment (guardrex) [inline aspnetcore/fundamentals/websockets.md:117]:
```suggestion
> If the compression negotiation isn't accepted by either the server or client, the connection is still established. However, the connection doesn't use compression when sending and receiving messages.
```

**Review by guardrex — APPROVED** [review]
I guess I should've just rolled the dice on a direct edit 😆.

File: aspnetcore/fundamentals/websockets.md
Comment (BrennanConroy) [reply aspnetcore/fundamentals/websockets.md:114]: What do you mean "negotatiation" isn't a word?!

File: aspnetcore/fundamentals/websockets.md
Comment (Rick-Anderson) [reply aspnetcore/fundamentals/websockets.md:97]: Or `Use compression"
Also, sentence casing.

File: aspnetcore/fundamentals/websockets.md
Comment (BrennanConroy) [reply aspnetcore/fundamentals/websockets.md:112]: This sounds off for some reason

Comment (guardrex) [reply aspnetcore/fundamentals/websockets.md:113]: It's the "of" bit. It could go ....

```suggestion
Additionally, <xref:Microsoft.AspNetCore.Http.WebSocketAcceptContext> advanced options, `ServerMaxWindowBits` and `DisableServerContextTakeover`, control how the compression works.
```

Comment (guardrex) [reply aspnetcore/fundamentals/websockets.md:113]: Also, this is a possibility ...

```suggestion
Additionally, `WebSocketAcceptContext.ServerMaxWindowBits` and `WebSocketAcceptContext.DisableServerContextTakeover` control how the compression works.
```

Comment (Rick-Anderson) [reply aspnetcore/fundamentals/websockets.md:113]: > This sounds off for some reason

agreed. Original is fine if they are really advanced, meaning most users won't use them.

Comment (Rick-Anderson) [reply aspnetcore/fundamentals/websockets.md:113]:
```suggestion
[ServerMaxWindowBits](TODO) and [DisableServerContextTakeover](TODO) are advanced options on [WebSocketAcceptContext](dotnet/api/microsoft.aspnetcore.http.websocketacceptcontext) that control how the compression works.
```

Comment (Rick-Anderson) [reply aspnetcore/fundamentals/websockets.md:113]: Additionally is dead wood.

Comment (guardrex) [reply aspnetcore/fundamentals/websockets.md:113]: The reason I suggested a change for the original was to avoid the state verb. I think my last suggestion is probably the best of the litter. 🐱

Comment (guardrex) [reply aspnetcore/fundamentals/websockets.md:113]:
```suggestion
`WebSocketAcceptContext.ServerMaxWindowBits` and `WebSocketAcceptContext.DisableServerContextTakeover` control how the compression works.
```

... using `?displayProperty=nameWithType` later when these to go API links.

Comment (BrennanConroy) [reply aspnetcore/fundamentals/websockets.md:113]:
```suggestion
`WebSocketAcceptContext.ServerMaxWindowBits` and `WebSocketAcceptContext.DisableServerContextTakeover` are advanced options that control how the compression works.
```

**Comment by BrennanConroy** [comment]
Want to wait for @blowdart to review before merging to make sure I have enough warnings in the doc 😃

File: aspnetcore/fundamentals/websockets.md
Comment (blowdart) [inline aspnetcore/fundamentals/websockets.md:100]: can make an app, rather than makes

**Review by blowdart — DISMISSED** [review]
Little tweek, because if your app is static text (hah, some app) then it's not vulnerable.

File: aspnetcore/fundamentals/websockets.md
Comment (guardrex) [reply aspnetcore/fundamentals/websockets.md:100]:
```suggestion
> Enabling compression over encrypted connections can make an app subject to CRIME/BREACH attacks.
```

**Pull request merged**
